### DISCUSSION TITLE
Is TinyTuya working without internet connectivity (after initial setup)?

### DISCUSSION THREAD

**Discussion by ricorico94** [jasonacox/tinytuya#344] [General]
Hi,
I'm not sure to understand.. Based on description of TinyTuya, it uses local network to manage (read & control) the local devices.
But at same time, it requires Cloud details for the setup at least.
So, I have 2 questions: 
1) once it is setup (I ave local keys, etc), does it need live internet connection to read status or control any local device ? (asking, because I need to be able to control some devices even if I lose my internet connection).
2) I see a comment about "how to extend after expiration" -> doe sit mean the local control of devices will stop working after XX months, with need to extend manually ? If yes, is there any way to be warned about expiration, to avoid it occurs during holidays for instance ?

Thanks a lot for your work

**Comment by uzlonewolf** [comment]
Hi @ricorico94 ,

1. It depends.  For most devices the answer is "no, it does not need a live internet connection," however a few devices (mainly cameras or other complex devices) have severely limited functionality when offline.
2. No, only connecting to the cloud to get the keys for new devices will stop working, devices you already have the keys for will continue to work until they are factory reset.

**Comment by ricorico94** [comment]
Sounds great, thanks a lot !
I'll probably replace your other "Tuya plugin" by this one, then ! Is there an easy way to migrate from one to the other ? (ie replacing the current devices in domoticz by the ones managed by TinyTuya)

**Reply by jasonacox** [reply]
Hi @ricorico94  - I don't know enough about domoticz to help.  You could check out @make-all's HA plugin at https://github.com/make-all/tuya-local

**Reply by make-all** [reply]
Domoticz is different than HA, I presume the plugin would be https://github.com/Xenomes/Domoticz-TinyTUYA-Plugin

**Reply by jasonacox** [reply]
Thanks, Jason!  I'll add that to our readme.

**Comment by heggink** [comment]
That one works for me quite well!
